Sources/MapboxNavigation/CarPlayNavigationViewController.swift

```diff
@@ -335,8 +335,10 @@ public class CarPlayNavigationViewController: UIViewController {
         }
         
         if routeLineTracksTraversal {
-            if routeProgress.isFinalLeg && routeProgress.currentLegProgress.distanceRemaining <= 0.0 {
-                navigationMapView?.removeRoutes()
+            if routeProgress.isFinalLeg && routeProgress.currentLegProgress.userHasArrivedAtWaypoint {
+                if routeProgress.currentLegProgress.distanceRemaining <= 0.0 {
```

Comment:
You could have all these checks in one `if` in correct order.

Reply:
I see this check repeats multiple times. Could we extract it into `RouteProgress` method or so?